mesonbuild/interpreterbase.py

```diff
@@ -55,6 +55,19 @@ def wrapped(self, node, args, kwargs):
         return f(self, node, args, kwargs)
     return wrapped
 
+class permittedKwargs:
+
+    def __init__(self, permitted):
+        self.permitted = permitted
+
+    def __call__(self, f):
+        def wrapped(s, node, args, kwargs):
+            for k in kwargs:
+                if k not in self.permitted:
+                    raise InvalidArguments('Invalid keyword argument %s.' % k)
```

Comment:
Warning here, please. Lots of people will have accidentally pushed broken kwargs into releases, and we should not break them.